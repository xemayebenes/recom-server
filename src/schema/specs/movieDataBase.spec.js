import {
    SearchMovieItem,
    SearchSerieItem,
    ItemInterface,
    Movie,
    Serie,
    VideoData,
    Season,
    Genre,
    Images
} from '../movieDataBase';

describe('SCHEMA MOVIE DATA BASE', () => {
    describe('SearchMovieItem', () => {
        it('should match snapshot', () => {
            expect(SearchMovieItem).toMatchSnapshot();
        });
    });
    describe('SearchSerieItem', () => {
        it('should match snapshot', () => {
            expect(SearchSerieItem).toMatchSnapshot();
        });
    });
    describe('ItemInterface', () => {
        it('should match snapshot', () => {
            expect(ItemInterface).toMatchSnapshot();
        });
    });
    describe('Movie', () => {
        it('should match snapshot', () => {
            expect(Movie).toMatchSnapshot();
        });
    });
    describe('Serie', () => {
        it('should match snapshot', () => {
            expect(Serie).toMatchSnapshot();
        });
    });
    describe('VideoData', () => {
        it('should match snapshot', () => {
            expect(VideoData).toMatchSnapshot();
        });
    });
    describe('Season', () => {
        it('should match snapshot', () => {
            expect(Season).toMatchSnapshot();
        });
    });
    describe('Genre', () => {
        it('should match snapshot', () => {
            expect(Genre).toMatchSnapshot();
        });
    });
    describe('Images', () => {
        it('should match snapshot', () => {
            expect(Images).toMatchSnapshot();
        });
    });
});