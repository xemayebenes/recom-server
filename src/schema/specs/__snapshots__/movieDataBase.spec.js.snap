// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCHEMA MOVIE DATA BASE Genre should match snapshot 1`] = `
"
  type Genre @cacheControl(maxAge: 5){
    id: Int!
    name: String
  }"
`;

exports[`SCHEMA MOVIE DATA BASE Images should match snapshot 1`] = `
"
type Images @cacheControl(maxAge: 5){
    main: String!
        secondary: String!
}"
`;

exports[`SCHEMA MOVIE DATA BASE ItemInterface should match snapshot 1`] = `
"interface ItemInterface {
    id: ID!
    externalId: Int!
    title: String!
    genres: [Genre]
    overview: String!
    popularity: Float
    vote_count: Int
    vote_average: Float
    images: Images!
  }"
`;

exports[`SCHEMA MOVIE DATA BASE Movie should match snapshot 1`] = `
"
  type Movie implements ItemInterface{
    id: ID!
    externalId: Int!
    title: String!
    genres: [Genre]
    overview: String!
    popularity: Float
    vote_count: Int
    vote_average: Float
    images: Images!
    imdb_id: String
    original_title: String!
    release_date: String
    video: Boolean
    images: Images!
    videoData: [VideoData]
    omdbData: FilmOMDBData
  }"
`;

exports[`SCHEMA MOVIE DATA BASE SearchMovieItem should match snapshot 1`] = `
"
type SearchMovieItem {
    externalId: Int!
    title: String
    images: Images!
    popularity: Float
    vote_average: Float
    overview: String!
    genre_ids: [Int]
    original_language: String
    vote_count: Int
    adult: Boolean
    release_date: String
    original_title: String
    video: Boolean
  }"
`;

exports[`SCHEMA MOVIE DATA BASE SearchSerieItem should match snapshot 1`] = `
"
  type SearchSerieItem {
    externalId: Int!
    title: String !
    images: Images!
    popularity: Float
    vote_average: Float
    overview: String!
    genre_ids: [Int]
    original_language: String
    vote_count: Int
    first_air_date: String
    origin_country: [String]  
  }"
`;

exports[`SCHEMA MOVIE DATA BASE Season should match snapshot 1`] = `
"
  type Season @cacheControl(maxAge: 5){
    id: Int
    air_date: String
    episode_count: Int
    poster_path: String
    season_number: Int
  }
  "
`;

exports[`SCHEMA MOVIE DATA BASE Serie should match snapshot 1`] = `
"
  type Serie implements ItemInterface @cacheControl(maxAge: 5){
    id: ID!
    externalId: Int!
    title: String!
    genres: [Genre]
    overview: String!
    popularity: Float
    vote_count: Int
    vote_average: Float
    images: Images!
    name: String !
    original_name: String
    number_of_episodes: Int
    number_of_seasons: Int
    seasons: [Season]
    videoData: [VideoData]
    omdbData: SerieOMDBData
  }"
`;

exports[`SCHEMA MOVIE DATA BASE VideoData should match snapshot 1`] = `
"
  type VideoData @cacheControl(maxAge: 5){
    id: String
    iso_639_1: String
    iso_3166_1: String
    key: String
    name: String
    site: String
    size: Int
    type: String
    trailer: String
  }"
`;
